app/store/chat.ts

```diff
@@ -81,9 +81,10 @@ function createEmptySession(): ChatSession {
   };
 }
 
-function getSummarizeModel(currentModel: string) {
-  // if it is using gpt-* models, force to use 3.5 to summarize
-  return currentModel.startsWith("gpt") ? SUMMARIZE_MODEL : currentModel;
+// fix known issue where summarize is not using the current model selected
+function getSummarizeModel(currentModel: string, modelConfig: ModelConfig) {
+  // should be depends of user selected
+  return currentModel ? modelConfig.model : currentModel;
```

Comment:
The initial intention was to use GPT-3.5 to save on token usage, as directly using the model currently in use by the user might significantly increase their usage costs.

Reply:
> The initial intention was to use GPT-3.5 to save on token usage, as directly using the model currently in use by the user might significantly increase their usage costs.

It's not about usage costs, in-scenario when you using model gpt-4 with latest knowledge 2023, and you are using old model just for summarizing conversation, it giving bad conversation for next chat when send memory is enabled example is confusing, not a connected about conversation

Reply:
be smart instead of thinking about cost

Reply:
There is no fundamental difference between GPT-3.5 and GPT-4 in generating a few words for a summary. Considering that some users have already raised concerns about costs, I believe it is still necessary to retain the use of GPT-3.5 for generating session titles

Reply:
> There is no fundamental difference between GPT-3.5 and GPT-4 in generating a few words for a summary. Considering that some users have already raised concerns about costs, I believe it is still necessary to retain the use of GPT-3.5 for generating session titles

it's not about session titles
why focusing about session titles which is useless because you can create masks in chats

here is note that you need to know:
in current unchanged it force to summarize of session memory include that useless chat/session titles using old models gpt3.5 and it's affected LMAO

Reply:
tokens usage are stable with this pr 

proof:

![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/f8ac0850-f395-4b43-9538-d9d36bde7b14)

Reply:
another note why its affected ?
"in-scenario when you using model gpt-4 with latest knowledge 2023, and you are using old model just for summarizing conversation, it giving bad conversation for next conversation when send memory is enabled example is confusing, not a connected about conversation"

Other:
Another issue with the conversation is related to the GPT-4-preview model, which has 128k context tokens, compared to the older model that only has 16k context tokens (for example). So, you can imagine how this scenario could lead to problems and inefficiencies.

Reply:
I'm referring to the cost rather than the number of tokens, the price of GPT-4 is significantly higher than that of GPT-3.5.

Reply:
The recommended way of action is to continue using GPT-3.5 Turbo when it's available, otherwise use the model currently enabled by the user.

Reply:
> The recommended way of action is to continue using GPT-3.5 Turbo when it's available, otherwise use the model currently enabled by the user.

if you think that is recommended
you have to close this issue 

- ChatGPTNextWeb#3621

for me tbh its not recommended